syntax = "proto3";

package seerep;

import "point_field.proto";
import "header.proto";
import "boundingbox_labeled.proto";

/**
 * This message holds a collection of N-dimensional points, which may
 * contain additional information such as normals, intensity, etc. The
 * point data is stored as a binary blob, its layout described by the
 * contents of the "fields" array.
 *
 * The point cloud data may be organized 2d (image-like) or 1d
 * (unordered). Point clouds organized as 2d images may be produced by
 * camera depth sensors such as stereo or time-of-flight.
 *
 * Time of sensor data acquisition, and the coordinate frame ID (for 3d
 * points).
 */

message PointCloud2
{
  Header header = 1;

  uint32 height = 2;
  uint32 width = 3;

  repeated PointField fields = 4;

  bool is_bigendian = 5;
  uint32 point_step = 6;
  uint32 row_step = 7;

  // the point cloud data
  bytes data = 8;
  bool is_dense = 9;

  // (optional) labels for the complete image
  repeated string labels_general = 10;

  // (optional) bounding box based labels
  repeated BoundingBoxLabeled labels_bb = 11;
}
